### PR TITLE
added support of item-template inside the dropdown

### LINKED ISSUE #250: Dropdown should support Item-template and label should also accpet a widget
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Need to add following support inside the dropdown
1. item template
2. label should support widget

## Issue ticket number and link

Closes #
closes #250 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Dropdown.tsx [hunk 1]
```diff
@@ -39,6 +63,41 @@ const Dropdown: React.FC<DropdownProps> = (props) => {
...
+    }
+
+    if (isObject(itemTemplate)) {
+      return namedData.map((item: unknown, index) => {
+        return (
+          <Select.Option
+            key={index}
+            value={evaluate(defaultScreenContext, itemTemplate.value, {
+              [itemTemplate.name]: get(item, itemTemplate.name) as unknown,
+            })}
```
Comment (sagardspeed2) [inline packages/runtime/src/widgets/Dropdown.tsx:90]: @evshi I need your help here,

as right now I am getting the error of 
```Unsafe assignment of an `any` value```

and If I wrap this with either Number or String, there is no error.
but we can't be sure that every time we will get Number or String as a value.

Comment (evshi) [reply packages/runtime/src/widgets/Dropdown.tsx:90]: `evaluate` takes a type argument that will case the return value

File: .changeset/tender-plums-crash.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@ensembleui/react-framework": patch
+"@ensembleui/react-runtime": patch
+---
+
+Fix the crashing of charts
```
Comment (evshi) [inline .changeset/tender-plums-crash.md:6]: Is this the right changeset?

File: apps/kitchen-sink/src/ensemble/screens/widgets.yaml [hunk 3]
```diff
@@ -54,10 +54,37 @@ View:
...
+                  item-template:
+                    data:
+                      [
+                        { name: "hello", value: 11 },
+                        { name: "buyyy", value: 22 },
+                      ]
+                    name: product
+                    value: ${product.value}
+                    template:
+                      Text:
```
Comment (evshi) [inline apps/kitchen-sink/src/ensemble/screens/widgets.yaml:68]: Does this follow the flutter sides contract? I think we also need to bind the value

File: packages/runtime/src/widgets/Dropdown.tsx [hunk 4]
```diff
@@ -1,34 +1,59 @@
...
 
 export type DropdownProps = {
   label?: string;
   hintText?: string;
   value?: Expression<string | number>;
-  items: SelectOption[];
+  items?: SelectOption[];
   onItemSelect: EnsembleAction;
   autoComplete: Expression<boolean>;
+  "item-template"?: DropdownItemTemplace;
```
Comment (evshi) [inline packages/runtime/src/widgets/Dropdown.tsx:38]: There is a shared type for this

File: packages/runtime/src/widgets/Dropdown.tsx [hunk 5]
```diff
@@ -39,6 +64,43 @@ const Dropdown: React.FC<DropdownProps> = (props) => {
...
     [action],
   );
+
+  const { namedData } = useTemplateData({
+    data: itemTemplate?.data,
+    name: itemTemplate?.name,
+  });
+
+  const options = useMemo(() => {
+    if (values?.items) {
```
Comment (evshi) [inline packages/runtime/src/widgets/Dropdown.tsx:74]: Can you confirm what the behavior is for flutter? What is the order of preference or perhaps they concatenate the options if both exist?

File: packages/runtime/src/widgets/Dropdown.tsx (hunk 4 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/Dropdown.tsx:38]: Yup, I know. but here we need extra args inside the `item-template`. so I used it like this

File: apps/kitchen-sink/src/ensemble/screens/widgets.yaml (hunk 3 above)
Comment (sagardspeed2) [reply apps/kitchen-sink/src/ensemble/screens/widgets.yaml:68]: Yes, I have seen the definition of flutter implementation for the item-template inside the dropdown. they are using this way to passing and evaluate the value on each option

File: packages/runtime/src/widgets/Dropdown.tsx (hunk 5 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/Dropdown.tsx:75]: It concatenate both, items and item-template if available, and priority is given to items. so as per that I updated the PR

File: packages/runtime/src/widgets/Dropdown.tsx (hunk 4 above)
Comment (evshi) [reply packages/runtime/src/widgets/Dropdown.tsx:38]: You can use type composition: `HasItemTemplate & { value: Expression<string> }`

**Review by evshi — COMMENTED** [review]
Looks good, just fix changeset and item template typing

**Review by evshi — APPROVED** [review]

**Pull request merged**